src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java

```diff
@@ -281,6 +261,26 @@ private void generateDiffPatchWithGitCommand(int headNum, String patchFormat) th
         }
     }
 
+    private void runShellCommand(String command) throws Exception {
+        final List<String> cmds = new ArrayList<>();
+        cmds.add("sh");
+        cmds.add("-c");
+        cmds.add(command);
+        final Process process = new ProcessBuilder()
+                .directory(new File(repoPath))
+                .command(cmds)
+                .inheritIO()
+                .start();
+        final int code = process.waitFor();
+        stateException(code, "an error occurred when running " + command);
```

Comment:
why exception not thrown right here? do you really separate method?

Reply:
done, yes it is not necessary to separate method.